src/dsp_tools/commands/xmlupload/models/value_transformers.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+import json
+from abc import ABC
+from abc import abstractmethod
+from typing import Any
+from typing import TypeAlias
+from typing import Union
+from typing import assert_never
+
+from dsp_tools.commands.xmlupload.models.formatted_text_value import FormattedTextValue
+from dsp_tools.models.exceptions import BaseError
+from dsp_tools.utils.date_util import Date
+from dsp_tools.utils.date_util import parse_date_string
+
+InputTypes: TypeAlias = Union[str, FormattedTextValue]
+OutputTypes: TypeAlias = Union[str, Date]
+
+
+class ValueTransformer(ABC):
+    """Class used to transform an input value."""
+
+    @abstractmethod
+    def transform(self, input_value: InputTypes) -> OutputTypes: ...
+
+    @staticmethod
+    def _assert_is_string(value: str | FormattedTextValue) -> str:
+        match value:
+            case str() as s:
+                return s
+            case FormattedTextValue() as xml:
+                raise BaseError(f"Expected string value, but got XML value: {xml.as_xml()}")
+            case _:
+                assert_never(value)
+
+
+class DateTransformer(ValueTransformer):
+    def transform(self, input_value: Any) -> Date:
+        val = self._assert_is_string(input_value)
+        return parse_date_string(val)
```

Comment:
This function is now only used here and in tests, perhaps we can in the future move it into this, but for the moment it works and keeps everything minimal.